script/spec-runner.js

```diff
@@ -38,8 +62,41 @@ function saveSpecHash ([newSpecHash, newSpecInstallHash]) {
 }
 
 async function runElectronTests () {
+  const errors = []
+  const runners = [
+    ['Remote based specs', 'remote', runRemoteBasedElectronTests],
+    ['Main process specs', 'main', runMainProcessElectronTests]
+  ]
+
+  const mochaFile = process.env.MOCHA_FILE
+  for (const runner of runners) {
+    if (only && !only.includes(runner[1])) {
+      console.info('\nSkipping:', runner[0])
+      continue
+    }
+    try {
+      console.info('\nRunning:', runner[0])
+      if (mochaFile) {
+        process.env.MOCHA_FILE = mochaFile.replace('.xml', `-${runner[1]}.xml`)
+      }
+      await runner[2]()
+    } catch (err) {
+      errors.push([runner[0], err])
+    }
+  }
+
+  if (errors.length !== 0) {
+    for (const err of errors) {
+      console.error('\n\nRunner Failed:', err[0])
+      console.error(err[1])
+    }
+    throw new Error('Electron test runners have failed')
+  }
+}
+
+async function runRemoteBasedElectronTests () {
   let exe = path.resolve(BASE, utils.getElectronExec())
-  const args = process.argv.slice(2)
+  const args = ['electron/spec', ...process.argv.slice(2).filter(arg => !arg.startsWith('--only='))]
```

Comment:
let's maybe be a little less gung ho about `argv`?

Reply:
We want to pass everything through here though so that spec runners get the arguments passed in